macros/CompatBeta.ejs

```diff
@@ -0,0 +1,603 @@
+      cssClass = 'partial';
+    }
+  }
+
+  return cssClass;
+}
+
+/*
+Generate the note for a browser flag or preference
+First checks version_added and version_removed to create a string indicating when
+a preference setting is present. Then creates a (browser specific) string
+for either a preference flag or a compile flag.
+
+// TODO handle array of flags (rename flag to flags)
+// see https://github.com/mdn/browser-compat-data/issues/546
+
+`supportData` is a support_statement
+`browserId` is a compat_block browser ID
+*/
+function writeFlagsNote(supportData, browserId) {
+  let output = '';
+
+  const firefoxPrefs = 'To change preferences in Firefox, visit about:config.';
+  const chromePrefs = 'To change preferences in Chrome, visit chrome://flags.';
+
+  if (typeof(supportData.version_added) === 'string') {
+    output = `From version ${supportData.version_added}`;
+  }
+
+  if (typeof(supportData.version_removed) === 'string') {
+    if (output) {
+      output += ` until version ${supportData.version_removed} (exclusive)`;
+    } else {
+      output = `Until version ${supportData.version_removed} (exclusive)`;
+    }
+  }
+
+  let flagTextStart = 'This';
+  if (output) {
+    output += ':';
+    flagTextStart = ' this';
+  }
+
+  let flagText = `${flagTextStart} feature is behind the <code>${supportData.flag.name}</code>`;
+
+  // value_to_set is optional
+  let valueToSet = '';
+  if (supportData.flag.value_to_set) {
+    valueToSet = ` (needs to be set to <code>${supportData.flag.value_to_set}</code>)`;
+  }
+
+  if (supportData.flag.type === 'preference') {
+    let prefSettings = '';
+    switch (browserId) {
+      case 'firefox':
+      case 'firefox_android':
+        prefSettings = firefoxPrefs;
+      break;
+      case 'chrome':
+      case 'chrome_android':
+        prefSettings = chromePrefs;
+      break;
+    }
+    output += `${flagText} preference${valueToSet}. ${prefSettings}`;
+  }
+
+  if (supportData.flag.type === 'compile_flag') {
+    output += `${flagText} compile flag${valueToSet}.`;
+  }
+
+  return output;
+}
+
+/*
+Generates icons for the main cell
+`supportData` is a support_statement
+
+*/
+function writeCellIcons(support) {
+  let output = '<div class="bc-icons">';
+
+  if (support.prefix) {
+    output += writeIcon('prefix', support.prefix) + ' ';
+  }
+
+  if (support.notes) {
+    output += writeIcon('footnote') + ' ';
+  }
+
+  if (support.alternative_name) {
+    output += writeIcon('altname', support.alternative_name) + ' ';
+  }
+
+  if (support.flag) {
+    // TODO handle array of flags (rename flag to flags)
+    // see https://github.com/mdn/browser-compat-data/issues/546
+    output += writeIcon('disabled') + ' ';
+  }
+
+  output += '</div>';
+
+  return output;
+}
+
+/*
+Create notes section
+
+`supportData` is a support_statement
+`browserId` is a compat_block browser ID
+
+*/
+function writeNotes(support, browserId) {
+  let output = '<section class="bc-history" aria-hidden="true"><dl>';
```

Comment:
Do you have any thoughts on changing this to an `<aside>`? Not sure I picked the right element here.

Reply:
`<aside>` might be better, but I'm really not an HTML or semantics expert :confused: